lib/recurly/resource.rb

```diff
@@ -608,7 +608,7 @@ def embedded!(root_index = false)
 
       def find_resource_class(name)
         resource_name = Helper.classify(name)
-        if Recurly.const_defined?(resource_name, false)
+        if Recurly.const_defined?(resource_name, false) && Recurly.const_get(resource_name, false).instance_of?(Class)
```

Comment:
can you explain what this is doing?

Reply:
There is both an attribute `version` on the `DunningCycle` resource and a class `Version` this additional check makes sure that when it is looking for the resource class that it is also verifying that the constant is for the class and not an attribute with the same name.